addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ExtensionOpenApi.java

```diff
@@ -343,9 +347,20 @@ public OpenApiResults importOpenApiDefinitionV2(
                 throw new InvalidDefinitionException();
             }
 
+            String openApiString;
+            try {
+                openApiString = Json.mapper().writeValueAsString(openApi);
```

Comment:
Using Json.mapper().writeValueAsString(openApi) instead of Json.pretty(openApi) as per explained in last comment to be able to catch the exception. Otherwise it is not possible as the underlying library simply prints the stacktrace.